dash/include/dash/Meta.h

```diff
@@ -90,6 +90,19 @@ struct is_container_compatible :
          >
 { };
 
+/**
+ * Type trait indicating whether a type can be used for global atomic
+ * operations.
+ */
+template <typename T>
+struct is_atomic_compatible
+: public std::integral_constant<
+           bool,
+              dash::is_container_compatible<T>::value
+           && sizeof(T) <= sizeof(std::size_t)
+         >
```

Comment:
We restrict to 64bit because MPI does not support wider types for by-value-atomic reduce operations. As a side-effect, atomic operations in local memory usually will be executed bia `CMPXCHG` or some other lock-free instruction. But: `std::is_lock_free` does not ensure that MPI supports reduce operations on that type.